tests/data/test_spark_dataset.py

```diff
@@ -282,15 +295,19 @@ def test_load_delta_path(spark_session, tmp_path, df):
 
 
 def test_load_delta_path_with_version(spark_session, tmp_path, df):
-    df_spark = spark_session.createDataFrame(df)
     path = str(tmp_path / "temp.delta")
-    df_spark.write.format("delta").mode("overwrite").save(path)
+
+    df_v0 = pd.DataFrame([[4, 5, 6], [4, 5, 6]], columns=["a", "b", "c"])
+    assert not df_v0.equals(df)
+    df_v0_spark = spark_session.createDataFrame(df_v0)
+    df_v0_spark.write.format("delta").mode("overwrite").save(path)
+
     # write again to create a new version
-    df_spark.write.format("delta").mode("overwrite").save(path)
```

Comment:
The previous test logic wrote the exact same DF twice. On my machine, this didn't create a new version and the test failed. Fixing it here as a drive-by.

Reply:
Thanks for catching and fixing this!